### PR TITLE
Change for Prometheus metrics naming

### PR DESCRIPTION
**Pull request by perosb**
**Description:**
Update docs for change in https://github.com/home-assistant/home-assistant/pull/24103

Also includes doc updates done in https://github.com/home-assistant/home-assistant.io/pull/9226

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

<a href="https://gitpod.io/#https://github.com/home-assistant/home-assistant.io/pull/9518"></a>

### CODE REVIEW

File: source/_components/prometheus.markdown [hunk 1]
```diff
@@ -20,44 +20,89 @@ To use the `prometheus` component in your installation, add the following to you
 ```yaml
 # Example configuration.yaml entry
 prometheus:
```
Comment (klaasnicolaas) [inline source/_components/prometheus.markdown]: Only required items in the first example

**Review by klaasnicolaas — DISMISSED** [review]

File: source/_components/prometheus.markdown [hunk 2]
```diff
@@ -23,43 +25,97 @@ prometheus:
...
 {% configuration %}
-namespace:
-  description: The "namespace" that will be assigned to all the Prometheus metrics. This is the prefix of the metric name. E.g., having `myhass` as the namespace will cause the device tracker metrics to be `myhass_device_tracker_state`, the switch metrics to be `myhass_switch_state` and so on. The default is to not add any prefix to the metrics name. (available in version 0.73.0 and later)
-  required: false
-  type: string
-filter:
-  description: Filtering directives for the integrations which should be included or excluded from recording.
-  required: false
-  type: list
+prometheus:
```
Comment (MartinHjelmare) [inline source/_components/prometheus.markdown]: We don't include the top level component domain name in the config spec.

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]

File: source/_components/prometheus.markdown [hunk 3]
```diff
@@ -23,43 +25,96 @@ prometheus:
...
-          required: false
-          type: list
-        domains:
-          description: The list of domains to be included from recordings.
-          required: false
-          type: list
+description: Prometheus configuration.
+required: true
+type: map
+keys:
```
Comment (MartinHjelmare) [inline source/_components/prometheus.markdown]: The above items should also be removed.

Comment (perosb) [reply source/_components/prometheus.markdown]: You're a fast reviewer ;)

**Review by MartinHjelmare — APPROVED** [review]

**Review by frenck — APPROVED** [review]
Thanks, @perosb! 👍

**Pull request merged**